File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""This module contains the definition of the class Rectangle
"""


class Rectangle:
    """Rectangle class
    """
    print_symbol = "#"

    def __init__(self, width=0, height=0):
        """Initialize the Rectangle object"""
        self.width = width
        self.height = height

    @property
    def width(self):
        """Returns the value of attribute 'width'"""
        return self.__width

    @width.setter
    def width(self, width):
        """Returns the value of attribute 'width'"""
        if type(width) is not int:
            raise TypeError("width must be an integer")
        if width < 0:
            raise ValueError("width must be >= 0")
        self.__width = width

    @property
    def height(self):
        """Returns the value of attribute 'height'"""
        return self.__height

    @height.setter
    def height(self, height):
        """Returns the value of attribute 'height'"""
        if type(height) is not int:
            raise TypeError("height must be an integer")
        if height < 0:
            raise ValueError("height must be >= 0")
        self.__height = height

    def area(self):
        """Returns the rectangle area"""
        return self.width * self.height

    def perimeter(self):
        """Returns the rectangle perimeter"""
        if (not self.height) or (not self.width):
            return 0
        return 2 * (self.height + self.width)

    def __str__(self):
        """Returns string representation of the Rectangle object"""
        s = ""
        if self.width and self.height:
            for i in range(self.height):
                s += self.width * str(Rectangle.print_symbol) + "\n"
        return s[:-1]
    pass

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node

const Rectangle = class Rectangle {
  constructor (w, h) {
    if (w && h && Number(w) > 0 && Number(h) > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    if (!this.height || !this.width) {
      return;
    }

    let out = '';
    for (let i = 0; i < this.height; i++) {
      for (let j = 0; j < this.width; j++) {
        out += 'X';
      }
      if (i < this.height - 1) {
        out += '\n';
      }
    }
    console.log(out);
  }
};

module.exports = Rectangle;
